central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -546,13 +547,29 @@ func (m *managerImpl) generateSingleReportFromWatcherResults(result *watcher.Sca
 	// Update ReportData
 	snapshot.ReportData = m.getReportData(result.ScanConfig)
 	// Add failed clusters to the report
+	clusterData := make(map[string]*report.ClusterData)
+	for _, cluster := range snapshot.ReportData.GetClusterStatus() {
+		data := &report.ClusterData{
+			ClusterId:   cluster.GetClusterId(),
+			ClusterName: cluster.GetClusterName(),
+		}
+		if failedCluster, ok := failedClusters[cluster.GetClusterId()]; ok {
+			failedCluster.ClusterName = cluster.GetClusterName()
+			data.FailedInfo = failedCluster
+		}
+		clusterData[cluster.GetClusterId()] = data
+	}
 	if len(failedClusters) > 0 {
```

Comment:
I think, there is no need for that `if`. If the `failedClusters` is empty, then the slice will be empty and the for loop would not execute. We can skip this, unless you want to - for example - add a log line in that branch.

Reply:
I did some refactors on that part. Now the logic is divided into `GetClusterData` and `addFailedClustersToTheSnapshot`.